src/AdminBarApi.php

```diff
@@ -7,36 +7,96 @@
  */
 class AdminBarApi extends AbstractEndpoint {
 
-	/**
-	 * Endpoint path
-	 *
-	 * @Override
-	 * @var String
-	 */
-	protected $endpoint = '/admin-bar';
-
-	/**
-	 * Get the data.
-	 *
-	 * @Override
-	 * @param \WP_REST_Request $request The request.
-	 *
-	 * @return array|\WP_Error
-	 */
-	public function endpoint_callback( \WP_REST_Request $request ) {
-		global $wp_admin_bar;
-
-		$current_user = wp_get_current_user();
-
-		$data = [
-			'site_name' => get_bloginfo( 'name' ),
-			'user_name' => $current_user->user_login,
-			'dashboard_url' => admin_url(),
-			'logout_url' => wp_logout_url(),
-			'edit_page_url' => '',
-			'widgets' => $wp_admin_bar->get_node( 'new-content' ),
-		];
-
-		return $this->filter_data( $data );
-	}
+  /**
+   * Endpoint path
+   *
+   * @Override
+   * @var String
+   */
+  protected $endpoint = '/admin-bar';
+
+  /**
+   * Cookie name
+   *
+   * @var String
+   */
+  protected $cookieName = 'wp_admin_bar_user'; 
+
+  /**
+   * Cookie path
+   *
+   * @var String
+   */
+  protected $cookiePath = ''; 
+
+  /**
+   * Cookie domain
+   *
+   * @var String
+   */
+  protected $cookieDomain = '';
+
+  /**
+   * Constructor
+   */
+  function __construct() {
+    add_action('wp_login', [ $this, 'login_action' ], 10, 1);
```

Comment:
I think we are going to use the default values for `priority`(10) and  `arguments` (1)  there is no need to specify this values: 

````php
add_action('wp_login', [ $this, 'login_action' ] 1);
```

In the same way you did on the next line.